src/link/linkconfiguration.h

```diff
@@ -24,6 +25,11 @@ class LinkConfiguration : public QObject
         QStringList args;
         QString name;
         LinkType type = LinkType::None;
+        //TODO: We should somehow make this class or this structure more abstract
+        // and remove any protocol or device specific information
+        // right now this is necessary, since link configuration is the our default
+        // link information structure
+        PingDeviceType deviceType = PingDeviceType::UNKNOWN;
```

Comment:
Can you remind me briefly why this is necessary @patrickelectric?

Reply:
This is necessary to share device identification from protocoldetector to devicemanager.